File: src/4node/classify/nodes/askForMoreNode.ts
// ─── askForMoreNode — 追问补全 ──────────────────────────────────────────────
// 归属：主图节点

import { HumanMessage, AIMessage } from '@langchain/core/messages'
import { interrupt } from '@langchain/langgraph'
import { createChatModel } from '../../../2model/model.js'
import { contextualAskPrompt } from '../../../1common/prompts.js'
import type { MainStateType } from '../../../3state/index.js'

export async function askForMoreNode(state: MainStateType): Promise<Partial<MainStateType>> {
  const allUserText = state.messages
    .filter(m => m instanceof HumanMessage)
    .map(m => String(m.content))
    .join('\n')

  const model = createChatModel(0)
  const reply = await model.invoke([new HumanMessage(contextualAskPrompt(allUserText, state.missingInfo))])
  const question = String(reply.content)

  console.log(`-main:askForMoreNode: ${question}`)

  const userAnswer = interrupt(question)
  return {
    messages: [
      new AIMessage(question),
      new HumanMessage(String(userAnswer)),
    ],
  }
}

File: src/nodes/classify/ask_for_more_node.py
"""askForMoreNode - 追问缺失信息 - 对应 TS: askForMoreNode.ts"""
from typing import Any
from langchain_core.messages import HumanMessage, AIMessage
from langchain_core.messages import BaseMessage
from models.llm import create_chat_model
from prompts.templates import contextual_ask_prompt


async def ask_for_more_node(state: dict) -> dict:
    """
    追问缺失信息节点
    
    归属: 主图节点
    功能: 当用户信息不足时，生成追问
    """
    messages: list[BaseMessage] = state.get("messages", [])
    missing_info: list[str] = state.get("missing_info", [])
    
    # 提取已有信息
    all_user_text = "\n".join([
        m.content for m in messages
        if isinstance(m, HumanMessage)
    ])
    
    # 生成追问
    model = create_chat_model(temperature=0.3)
    
    prompt = contextual_ask_prompt(all_user_text, missing_info)
    response = await model.ainvoke([HumanMessage(content=prompt)])
    
    ask_message = response.content if hasattr(response, 'content') else str(response)
    
    print(f"-main:askForMoreNode: 追问={ask_message[:50]}...")
    
    return {
        "messages": [
            AIMessage(content=ask_message)
        ]
    }
